src/project/projectService.ts

```diff
@@ -23,18 +23,67 @@ import type {
   ProjectDeleteMutationInput,
   ProjectUpdateMutationInput,
 } from 'terraso-client-shared/graphqlSchema/graphql';
-import type { Project } from 'terraso-client-shared/project/projectSlice';
+import { Membership } from 'terraso-client-shared/memberships/membershipsSlice';
+import type {
+  HydratedProject,
+  Project,
+  SerializableSet,
+} from 'terraso-client-shared/project/projectSlice';
+import { Site } from 'terraso-client-shared/site/siteSlice';
 import * as terrasoApi from 'terraso-client-shared/terrasoApi/api';
 import {
   collapseConnectionEdges,
   collapseFields,
 } from 'terraso-client-shared/terrasoApi/utils';
 
-const collapseProjectFields = collapseFields<ProjectDataFragment, Project>({
-  userCount: inp => inp['group']['memberships']['totalCount'],
-  updatedAt: inp => new Date(inp['updatedAt']).toLocaleDateString(),
-  siteCount: inp => inp['siteSet']['totalCount'],
-});
+const collapseProjectFields = collapseFields<
+  ProjectDataFragment,
+  HydratedProject
+>(
+  {
+    dehydrated: inp => {
+      const siteIds = inp.siteSet.edges
+        .map(edge => edge.node.id)
+        .reduce((x, y) => ({ ...x, [y]: true }), {} as SerializableSet);
+      const membershipIds = inp.group.memberships.edges
+        .map(edge => edge.node)
+        .reduce(
+          (x, { id, user }) => ({ ...x, [id]: { user: user.id } }),
+          {} as Record<string, { user: string }>,
+        );
+      const output: Project = {
+        ...inp,
+        siteIds,
+        membershipIds,
+        groupId: inp.group.id,
+      };
+      return output;
+    },
+    sites: inp =>
```

Comment:
Question: is `inp` the project?
Suggestion: I think we need a better name for `inp` to improve readability